### ISSUE TITLE
[Bug] [Linux] Local World Persistance Does Not Work

### ISSUE THREAD

**Issue by CorneliusCornbread** [vrcx-team/VRCX#827]
**Describe the bug**
Whenever I join a world that uses VRCX's ability to save information from VRChat it does not work like it does on Windows.

**To Reproduce**
1. Go to a world that supports VRCX saving (e.g. Idle Home or YTS 2.1)
2. Wait a few seconds
3. World will not be able to load data

**Expected behavior**
It'd be nice if this feature worked on Linux just like it does on Windows.

**What version you are running**
Version 2024.06.12

~~I tried to see if there was a newer build, and redownload the latest artifact to verify I had the latest version. Current build is so out of date that the artifact is expired on GH :(~~
![image](https://github.com/vrcx-team/VRCX/assets/8294697/ee82d8d7-8760-44c9-b84d-0d06e801d43b)

**Comment by Myrkie** [comment]
VRCX doesn't use OSC in capacity, world persistence is based on a server and log reading, make sure that port `22500` is not being used, you should also be installing vrcx through this method https://github.com/vrcx-team/VRCX/tree/master/Linux as its more stable

**Comment by CorneliusCornbread** [comment]
> VRCX doesn't use OSC in capacity, world persistence is based on a server and log reading, make sure that port `22500` is not being used, you should also be installing vrcx through this method https://github.com/vrcx-team/VRCX/tree/master/Linux as its more stable

Installing this way just has the installer fail silently without saying much of anything

```
warning: Running /usr/bin/wineserver -w. This will hang until all wine processes in prefix=/home/jack/.local/share/vrcx terminate
------------------------------------------------------
Executing wine C:\windows\syswow64\regedit.exe /S C:\windows\Temp\_register-font.reg
002c:fixme:winediag:loader_init wine-staging 9.5 is a testing version containing experimental patches.
002c:fixme:winediag:loader_init Please mention your exact version when filing bug reports on winehq.org.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
0088:fixme:wineusb:query_id Unhandled ID query type 0x5.
Executing wine64 C:\windows\regedit.exe /S C:\windows\Temp\_register-font.reg
Executing cp /home/jack/.local/share/vrcx/dosdevices/c:/windows/temp/_register-font.reg /tmp/winetricks.LGe4abv9/_reg_405b24a4_66045.reg
Executing wine C:\windows\syswow64\regedit.exe /S C:\windows\Temp\_register-font.reg
Executing wine64 C:\windows\regedit.exe /S C:\windows\Temp\_register-font.reg
Executing cp /home/jack/.local/share/vrcx/dosdevices/c:/windows/temp/_register-font.reg /tmp/winetricks.LGe4abv9/_reg_70fc14d8_66045.reg
Download VRCX
Unknown option 'show-progress'
```

**Comment by Myrkie** [comment]
you might want to update wget that error is coming from this [part of thescript,](https://github.com/vrcx-team/VRCX/blob/master/Linux/install-vrcx.sh#L60-L67) which means your wget version is [too old](https://github.com/pswietojanski/slurp/issues/1#issuecomment-765926305) you should try updating with `apt-get install --only-upgrade wget`

**Comment by CorneliusCornbread** [comment]
> you might want to update wget that error is coming from this [part of the script](https://github.com/vrcx-team/VRCX/blob/master/Linux/install-vrcx.sh#L60-L67) which means your wget version is [too old](https://github.com/pswietojanski/slurp/issues/1#issuecomment-765926305) you should try updating with `apt-get install --only-upgrade wget`

im running the latest version of fedora 40 which has wget 2.1.0

**Comment by Myrkie** [comment]
if that's the case then the version in your distros repository doesn't support it, you could modify the script and remove all references to the argument `--show-progress`

**Comment by regalialong** [comment]
Seems like that Fedora has some weirdness with how they handle [progress related flags](https://src.fedoraproject.org/rpms/wget2/c/ff3ad5a501070454ea53915644b52dedb89dc40c). Withholding my annoyance, maybe it's worth replacing that with curl instead.

**Comment by Natsumi-sama** [comment]
If you'd like to debug the crash you can hit `http://127.0.0.1:22500/vrcx/data/getall` otherwise to disable the feature `Settings -> Advanced -> Disable Local World Persistence`

**Comment by CorneliusCornbread** [comment]
I downloaded the script and removed all instances of `--show-progress` and was able to get VRCX 2024.06.12 working and installed. But it seems like the local persistence doesn't work at all. Idle home reports that "Manual save was not found" when attempting to load from VRCX

**Comment by regalialong** [comment]
Does `curl http://127.0.0.1:22500/vrcx/data/getall` connect to anything?

**Comment by CorneliusCornbread** [comment]
> Does `curl http://127.0.0.1:22500/vrcx/data/getall` connect to anything?

Yes, it returns the following
```
{"ok":true,"error":null,"data":"{}","statusCode":200,"connectionKey":"KEY-HERE"}% 
```

**Comment by Natsumi-sama** [comment]
As long as the gameLog tab is working correctly that's all there is to it, maybe you haven't saved in this world yet.

**Comment by CorneliusCornbread** [comment]
> As long as the gameLog tab is working correctly that's all there is to it, maybe you haven't saved in this world yet.

Nah, it's full greyed out on Idle Home meaning it's not being detected and YTS 2.1 autosaves whenever you change but the autosave appears to not be working at all.

**Comment by regalialong** [comment]
Not able to repro this.
![](https://github.com/vrcx-team/VRCX/assets/126194895/cc00f8a9-2eb1-42a1-ab6a-700c0fdc0539)

**Comment by CorneliusCornbread** [comment]
> Not able to repro this. ![](https://private-user-images.githubusercontent.com/126194895/342756317-cc00f8a9-2eb1-42a1-ab6a-700c0fdc0539.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzNjk1NjAsIm5iZiI6MTcxOTM2OTI2MCwicGF0aCI6Ii8xMjYxOTQ4OTUvMzQyNzU2MzE3LWNjMDBmOGE5LTJlYjEtNDJhMS1hYjZhLTcwMGMwZmRjMDUzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQwMjM0MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMmUwNjg2YTMyZjE3MTNiNGQyYWUwYWI1ZTIwYjQ0NjY5Njc4ODE0NjI0YmRmMzBkNDI4NjNiZmIyZmM4M2M2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RDZyvTIIi6l-Rt7im51O2pI_1sutMN0pUn5GgqioWV4)

I can reproduce this message in YTS, however, nothing is ever actually saved and none of my settings are actually properly loaded. For example, I turn off chairs and enable colliders, this saves just fine on Windows, but on Linux it doesn't actually function. So it can communicate just fine to VRCX but VRCX isn't actually saving anything.

**Comment by GroovyTeacup** [comment]
Can you post your VRCX.log in `%APPDATA%/VRCX/logs`? After a session where you try to use PWI functionality.

**Comment by CorneliusCornbread** [comment]
> Can you post your VRCX.log in `%APPDATA%/VRCX/logs`? After a session where you try to use PWI functionality. EDIT: Forgot you're on linux; Where your VRCX DB would be located there'll be a logs folder lol

```
2024-06-26 02:53:06.9812 [INFO ] VRCX                 - VRCX 2024.06.12 Starting... 
2024-06-26 02:53:07.0121 [DEBUG] VRCX.ProcessMonitor  - Added process vrchat to process monitor. 
2024-06-26 02:53:07.0129 [DEBUG] VRCX.ProcessMonitor  - Added process vrserver to process monitor. 
2024-06-26 02:53:07.2484 [INFO ] VRCX.WorldDBManager  - Listening for requests on http://127.0.0.1:22500/ 
2024-06-26 02:53:11.3268 [INFO ] VRCX                 - SetDateTill: 6/26/2024 2:53:11 AM 
2024-06-26 02:53:14.0270 [DEBUG] Javascript           - Fetched group Secret Clink (file://vrcx/app.js:7) 
2024-06-26 02:53:42.4739 [DEBUG] VRCX.WorldDBManager  - Received a request to 'http://127.0.0.1:22500/vrcx/data/getall' 
2024-06-26 02:53:42.6126 [DEBUG] Javascript           - PWI: location didn't match, fetched user location wrld_900dd077-1337-c0fe-babe-71de05ea12c4:91265~private(usr_dbdddc50-6005-4b6d-862c-8c2f46d97665)~region(use) (file://vrcx/app.js:7) 
2024-06-26 02:53:42.6126 [DEBUG] Javascript           - PWI: returning user location wrld_900dd077-1337-c0fe-babe-71de05ea12c4:91265~private(usr_dbdddc50-6005-4b6d-862c-8c2f46d97665)~region(use) (file://vrcx/app.js:7) 
2024-06-26 02:53:42.6437 [DEBUG] VRCX.WorldDBManager  - Serving a request for all data (0 entries) for world ID 'wrld_900dd077-1337-c0fe-babe-71de05ea12c4' with connection key KEY_HERE. 
```

**Comment by GroovyTeacup** [comment]
Can you repeat in a world with VRCX functionality? One of the ones you're trying to use.

**Comment by CorneliusCornbread** [comment]
> Can you repeat in a world with VRCX functionality? One of the ones you're trying to use.

I did, I repeated it in YTS 2.1, that's the log from that world

**Comment by GroovyTeacup** [comment]
It's pretty unusual that it's trying to get data with an invalid connection key that's not defined in VRCX/pwimanager so maybe a placeholder by the world. Should've added more logging for when connection keys are generated...

Can you attempt to save data in the idle world/yts and send the relevant portions of your VRChat log? If they're using the default impl, that'll give me a better idea of what's going on. 

Though before doing so, please upload your `VRCX-WorldData.db` and then delete/rename the file so it can re-initialize the worlds you're trying to use.

**Comment by natarii** [comment]
@CorneliusCornbread If you search your vrchat client logs for `[YTS] ConfManager: VRCX PWI connection key`, that shows verbatim the key that was returned from the initial `/vrcx/data/getall` request, might be helpful for debugging

**Comment by CorneliusCornbread** [comment]
> It's pretty unusual that it's trying to get data with an invalid connection key that's not defined in VRCX/pwimanager so maybe a placeholder by the world. Should've added more logging for when connection keys are generated...
> 
> Can you attempt to save data in the idle world/yts and send the relevant portions of your VRChat log? If they're using the default impl, that'll give me a better idea of what's going on.
> 
> Though before doing so, please upload your `VRCX-WorldData.db` so I can see what's going on there and then delete/rename the file so it can re-initialize the worlds you're trying to use.

The key being weird is actually just me being over-cautious and removing the key from the logs. If it's not sensitive information I can include it in future

Here's another set of logs of me trying to get the VRCX data to load in YTS
```
2024-06-26 22:27:43.5644 [INFO ] VRCX                 - VRCX 2024.06.12 Starting... 
2024-06-26 22:27:43.6268 [DEBUG] VRCX.ProcessMonitor  - Added process vrchat to process monitor. 
2024-06-26 22:27:43.6281 [DEBUG] VRCX.ProcessMonitor  - Added process vrserver to process monitor. 
2024-06-26 22:27:44.0806 [INFO ] VRCX.WorldDBManager  - Listening for requests on http://127.0.0.1:22500/ 
2024-06-26 22:27:50.7327 [INFO ] VRCX                 - SetDateTill: 6/26/2024 10:27:50 PM 
2024-06-26 22:31:13.1122 [DEBUG] VRCX.WorldDBManager  - Received a request to 'http://127.0.0.1:22500/vrcx/data/getall' 
2024-06-26 22:31:13.2887 [DEBUG] Javascript           - PWI: location didn't match, fetched user location wrld_900dd077-1337-c0fe-babe-71de05ea12c4:49593~private(usr_dbdddc50-6005-4b6d-862c-8c2f46d97665)~region(use) (file://vrcx/app.js:7) 
2024-06-26 22:31:13.2887 [DEBUG] Javascript           - PWI: returning user location wrld_900dd077-1337-c0fe-babe-71de05ea12c4:49593~private(usr_dbdddc50-6005-4b6d-862c-8c2f46d97665)~region(use) (file://vrcx/app.js:7) 
2024-06-26 22:31:13.3194 [DEBUG] VRCX.WorldDBManager  - Serving a request for all data (0 entries) for world ID 'wrld_900dd077-1337-c0fe-babe-71de05ea12c4' with connection key 9c7a01a8-6e26-416e-9a96-dc700abe552a. 
2024-06-26 22:31:44.1204 [DEBUG] VRCX.WorldDBManager  - Received a request to 'http://127.0.0.1:22500/vrcx/data/getall' 
2024-06-26 22:31:44.3114 [DEBUG] Javascript           - PWI: location didn't match, fetched user location wrld_900dd077-1337-c0fe-babe-71de05ea12c4:49593~private(usr_dbdddc50-6005-4b6d-862c-8c2f46d97665)~region(use) (file://vrcx/app.js:7) 
2024-06-26 22:31:44.3114 [DEBUG] Javascript           - PWI: returning user location wrld_900dd077-1337-c0fe-babe-71de05ea12c4:49593~private(usr_dbdddc50-6005-4b6d-862c-8c2f46d97665)~region(use) (file://vrcx/app.js:7) 
2024-06-26 22:31:44.3139 [DEBUG] VRCX.WorldDBManager  - Serving a request for all data (0 entries) for world ID 'wrld_900dd077-1337-c0fe-babe-71de05ea12c4' with connection key 9c7a01a8-6e26-416e-9a96-dc700abe552a. 
```

With the database file (I had to compress it as GH wouldn't accept the file type otherwise):  
[VRCX-WorldData.zip](https://github.com/user-attachments/files/16007598/VRCX-WorldData.zip)

**Comment by Natsumi-sama** [comment]
Are you *sure* your gameLog tab is working? because it really doesn't look like it.

**Comment by CorneliusCornbread** [comment]
> Are you _sure_ your gameLog tab is working? because it really doesn't look like it.

Ahh, I misunderstood you initially, my bad. Uhh, it's kinda working? It feels like there should be more entries to this list but there's not.

![image](https://github.com/vrcx-team/VRCX/assets/8294697/7472e9c3-efd7-4a1c-9607-2706efb1630e)

**Comment by CorneliusCornbread** [comment]
So I did do a minor workaround yesterday as for some reason there were too VRCX's in KDE for me, one was listed as a "Application" and the other was listed as a "wine application." They were two different installs of VRCX in the same wine prefix, for some reason the older *initial* install (2024.05.something I can't remember) was installed to `C:/vrcx` where the new version (2024.06.12) was installed to `C:/Program Files/vrcx`. It was annoying opening the wrong VRCX, especially with autostart so I deleted `C:/vrcx` and symlinked it to the program files vrcx just in case. But even before I was making sure I was opening the newer VRCX even after autostart?

TL;DR I have little to no idea why it's working, but maybe the installer is installing VRCX to a weird path but the updater is updating it to the regular Windows install path?
